fs/fd.h

```diff
@@ -55,6 +55,14 @@ struct fd {
             // locked by fd->lock
             struct list unix_scm;
         } socket;
+
+        struct {
+            uint64_t generation;
+            void* wbuffer;
+            size_t wbuffer_off;
```

Comment:
Why is this separate from fd->offset?

Reply:
See https://github.com/tbodt/ish/pull/447#issuecomment-516057341 -- it's offset subtracted from fd->offset to get offset into wbuffer.